### PR TITLE
Account Provisioning API

### PR DESCRIPTION
**Pull request by RTLcoil**
Added support for the account provisioning API.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -209,6 +210,22 @@ def ==(other)
   end
 end
 
+RSpec::Matchers.define :have_cloudinary_config do |expected|
+  match do |config|
+    [:cloud_name, :api_key, :api_secret].reduce(true) do |result, config_name|
+      result && (config.public_send(config_name) == expected[config_name])
+    end
```
Comment (amcaplan) [inline spec/spec_helper.rb:217]:
```suggestion
    [:cloud_name, :api_key, :api_secret].all? do |config_name|
      config.public_send(config_name) == expected[config_name]
    end
```

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -209,6 +210,22 @@ def ==(other)
...
+  end
+end
+
+RSpec::Matchers.define :have_cloudinary_account_config do |expected|
+  match do |config|
+    [:account_id, :provisioning_api_key, :provisioning_api_secret].reduce(true) do |result, config_name|
+      result && (config.public_send(config_name) == expected[config_name])
+    end
```
Comment (amcaplan) [inline spec/spec_helper.rb:225]:
```suggestion
    [:account_id, :provisioning_api_key, :provisioning_api_secret].all? do |config_name|
      config.public_send(config_name) == expected[config_name]
    end
```

**Comment by RTLcoil** [comment]
@amcaplan Good idea. Suggested changes applied.

The CI tests fail now but it looks like an unrelated issue (`Cloudinary::BaseApi::BadRequest: Cannot have more than 200 items`).

I reran [one of the older commits in Travis](https://travis-ci.org/github/cloudinary/cloudinary_gem/builds/705728425) and it now breaks with the same error as the latest commits... this is a commit that passed successfully last time it ran.

FYI: @const-cloudinary

**Comment by RTLcoil** [comment]
@amcaplan 
Merged from _master_. The tests pass now.

File: lib/cloudinary/base_config.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
+module Cloudinary
+  class BaseConfig
```
Comment (const-cloudinary) [inline lib/cloudinary/base_config.rb:2]: In my IDE (RubyMine) all config properties are underlined with warning, saying: 
`Cannot find [property name]`. 
Those warnings need to be fixed.

Overall a very good improvement!

**Pull request merged**